### PR TITLE
[improvement](tablet scheduler) fix higher priority tablet add failed due to pending queue full

### PR DESCRIPTION
**Pull request by yujun777**
When tablet checker search for unhealth tablets,  it will search table one by one.  If the pending queue is full,  then the later search tablets couldn't add into the pending queue,  even if they are higher priorty.  So we use a minmax queue to holding the tablets. If a tablet is higher priority and the pending queue is full,  it will envit the lowest priority tablet in the queue, then add this tablet into queue;

other:
1. increase mow version incomplete task sched priority;

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/clone/ColocateTableCheckerAndBalancer.java [hunk 1]
```diff
@@ -561,14 +561,14 @@ private void matchGroups() {
                                     tabletCtx.setIsUniqKeyMergeOnWrite(isUniqKeyMergeOnWrite);
 
                                     AddResult res = tabletScheduler.addTablet(tabletCtx, false /* not force */);
-                                    if (res == AddResult.LIMIT_EXCEED || res == AddResult.DISABLED) {
+                                    if (res == AddResult.DISABLED) {
```
Comment (deardeng) [inline fe/fe-core/src/main/java/org/apache/doris/clone/ColocateTableCheckerAndBalancer.java:564]: LIMIT_EXCEED 状态没处理，看需要加log不

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletChecker.java [hunk 2]
```diff
@@ -404,7 +404,7 @@ private LoopControlStatus handlePartitionTablet(Database db, OlapTable tbl, Part
                 tabletCtx.setIsUniqKeyMergeOnWrite(isUniqKeyMergeOnWrite);
 
                 AddResult res = tabletScheduler.addTablet(tabletCtx, false /* not force */);
-                if (res == AddResult.LIMIT_EXCEED || res == AddResult.DISABLED) {
+                if (res == AddResult.DISABLED) {
```
Comment (deardeng) [inline fe/fe-core/src/main/java/org/apache/doris/clone/TabletChecker.java:407]: ditto

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java [hunk 3]
```diff
@@ -277,9 +277,15 @@ public synchronized AddResult addTablet(TabletSchedCtx tablet, boolean force) {
         // if this is not a force add,
         // and number of scheduling tablets exceed the limit,
         // refuse to add.
-        if (!force && (pendingTablets.size() > Config.max_scheduling_tablets
-                || runningTablets.size() > Config.max_scheduling_tablets)) {
-            return AddResult.LIMIT_EXCEED;
+        if (!force && (pendingTablets.size() >= Config.max_scheduling_tablets
+                || runningTablets.size() >= Config.max_scheduling_tablets)) {
+            TabletSchedCtx lowestPriorityTablet = pendingTablets.peekLast();
+            if (lowestPriorityTablet == null || lowestPriorityTablet.compareTo(tablet) <= 0) {
```
Comment (deardeng) [inline fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java:283]: add comment

File: fe/fe-core/src/main/java/org/apache/doris/clone/ColocateTableCheckerAndBalancer.java (hunk 1 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/clone/ColocateTableCheckerAndBalancer.java:564]: 需要repair 的tablet 很多时， 这个limit 可能会很多日志。这里把limit exceed 的计数给打印出来了

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java (hunk 3 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java:287]: added

**Comment by yujun777** [comment]
run buildall

**Review by deardeng — APPROVED** [review]
LGTM

**Comment by yujun777** [comment]
run p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**